base_tier_validation/models/tier_validation.py

```diff
@@ -35,11 +35,26 @@ class TierValidation(models.AbstractModel):
         search="_search_reviewer_ids",
     )
     can_review = fields.Boolean(compute="_compute_can_review")
+    has_comment = fields.Boolean(
+        related='review_ids.has_comment',
+    )
+    approve_level = fields.Boolean(
+        related='review_ids.approve_level',
+    )
 
     @api.multi
     def _compute_can_review(self):
         for rec in self:
             rec.can_review = self.env.user in rec.reviewer_ids
+            if rec.can_review and rec.approve_level:
```

Comment:
Overall, can this logic become shorter? I looks quite complicated to me.

Reply:
`rec.reviewer_ids` and `r.reviewer_ids` are different.
- `rec.reviewer_ids`  are reviewer of document.
- `r.reviewer_ids` are reviewer of `rec.review_ids`.